### PR TITLE
Add openpyxl support to XLSX renderer

### LINKED ISSUE #121: Add openpyxl support to XLSX renderer
Once blazelibs/blazeutils#18 is addressed, we'll need to update the XLSX renderer to support openpyxl, since WriterX will default to using openpyxl if installed. The XLSX renderer is currently specific to xlsxwriter

### PR DESCRIPTION
**Pull request by pytrumpeter**
Fixes #121 

Based on changes for #124

### CODE REVIEW

File: webgrid/renderers.py [hunk 1]
```diff
@@ -1219,166 +1214,151 @@ def get_add_filter_row(self):
...
+    breaking changes.
 
-        return wb
+    Dispatches a workbook class depending on what libraries are available.
+    """
 
-    def sheet_header(self, xlh):
-        pass
+    def __new__(cls, *args, **kwargs):
+        if openpyxl and openpyxl.__class__.__name__ == 'module':
```
Comment (guruofgentoo) [inline webgrid/renderers.py:1257]: I think at this point we'd want xlsxwriter usage relegated to backwards-compatibility. I think blazeutils defaults to using openpyxl if it is available. We could do the same here for now, but we'd need a clear way for a dev to manually make the renderer use xlsxwriter if they have a report already written that uses it.

File: webgrid/renderers.py [hunk 2]
```diff
@@ -1219,166 +1214,151 @@ def get_add_filter_row(self):
...
+            return super().__new__(cls)
+        if xlsxwriter:
+            return xlsxwriter_workbook_factory(*args, **kwargs)
+        raise ImportError('You must have either openpyxl or xlsxwriter installed.')
+
+    def __init__(self, *args, **kwargs):
+        self.use_openpyxl = True
```
Comment (guruofgentoo) [inline webgrid/renderers.py:1264]: I see a lot of `if` blocks using this to determine which lib's functionality to use. I would rather those kinds of overrides be built as subclasses of a common interface. I.e. have a common sheet library interface class, subclass it twice for openpyxl and xlsxwriter. Determine which one is needed once at class init, then these if-blocks become method calls to that helper object. That would simplify what's going on here IMO.

**Review by guruofgentoo — CHANGES_REQUESTED** [review]
What's here works, but we should reorganize it a bit to make it a) more maintainable over time, b) easier to add another sheet-writing library in the future, and c) easier to remove xlsxwriter support if we ever want to.

File: webgrid/renderers.py (hunk 2 above)
Comment (pytrumpeter) [reply webgrid/renderers.py:1264]: I got a little ways into that approach with the `WorkbookBase` class but was concerned about breaking overrides that might exist elsewhere for the renderer's API.  Do you know of anywhere that the XLSX renderer itself is subclassed?

Comment (guruofgentoo) [reply webgrid/renderers.py:1264]: I'm not aware of any in active projects.

File: webgrid/__init__.py [hunk 3]
```diff
@@ -807,7 +763,7 @@ class ColumnGroup(object):
 
     Args:
         label (str): Grouping label to be rendered for the column set.
-        class\_ (str): CSS class name to apply in HTML rendering.
+        class_ (str): CSS class name to apply in HTML rendering.
```
Comment (guruofgentoo) [inline webgrid/__init__.py:766]: I'm pretty sure the \ escape was needed for something. Check the sphinx output against https://webgrid.readthedocs.io/en/latest/columns/base-column.html#webgrid.ColumnGroup

That made me look to see if we have a docs check in webgrid, like https://github.com/level12/keg-auth/blob/master/tox.ini#L58, and we don't... It needs to be there; please add that block for tox and make sure it passes.

**Review by guruofgentoo — CHANGES_REQUESTED** [review]
From a code arch perspective, this is much much better. FWIW anytime we see the same conditional (or similar conditions) over and over again, we need to take a step back and refactor that. Having them all verbose throughout the code makes all of the code responsible to know the difference between the cases, where that can often be abstracted.

File: webgrid/__init__.py (hunk 3 above)
Comment (pytrumpeter) [reply webgrid/__init__.py]: Unsurprisingly, adding the docs check made it fail, so I put the escapes back.

**Pull request merged**